lib/studies/shrine/app.dart

```diff
@@ -196,9 +196,12 @@ class _RestorableAppStateModel extends RestorableListenable<AppStateModel> {
 
     // Reset cart items.
     final cartItems = appData['cart_data'] as Map<dynamic, dynamic>;
-    cartItems.forEach((dynamic id, dynamic quantity) {
+    final cartEntries = cartItems.entries.toList();
```

Comment:
Please use `for (final cart in cartItems) { ... }` as this approach is verbose and error prone.